master/flock_server/static/flock-mpi.js

```diff
@@ -508,3 +510,50 @@ mpi.ireduce = async function (sendArr, op, comm) {
         return local;
     }
 }
+
+
+/**
+ * Each process sends the contents of its sendbuffer to the root process. The root process
+ * receives the messages and stores them in rank order.
+ * 
+ * @async
+ * @function mpi.igather
+ * 
+ * @param {array} sendArr - Array to 
+ * @param {function} root - The rank of the root node for the gather operation
+ * @param {string} comm - Name of the communication group to operate under
+ * 
+ * @returns {serializable} - Result of the gather is given to the node with rank === 0, other nodes will receive error codes
+ * 
+ */
+mpi.igather = async function (sendArr, root, comm) {
+
+    const TAG = 'internal_igather';
+        
+    let size = await mpi.getSize(comm);
+    let rank = await mpi.getRank(comm);
+    
+    if (rank === root) {
+
+        if (!Array.isArray(sendArr)) {
```

Comment:
Is this leftover from iscatter (based on the message)? 

Also, I would think there aren't requirements on the sendArr argument on the root node, but I'm not sure if there is another reason to check.

Reply:
If we don't require it to be an array, should we just gather into a new variable and return that?